src/stringParamAllocator.js

```diff
@@ -0,0 +1,92 @@
+import { isArrayAllStrings } from './validation.js';
+
+/* global Module */
+export default class ParamParser {
+    constructor(args, errorHandling) {
+        let self = this;
+
+        if (!isArrayAllStrings(args)) {
+            throw new Error('All items in the argument list must be strings');
+        }
+        // So first, we need to allocate Emscripten heap space sufficient to store each string as a
+        // null-terminated C string.
+        // Because the C function signature is char **, this array of pointers is going to need to
+        // get copied into Emscripten heap space eventually, so we're going to prepare by storing
+        // the pointers as a typed array so that we can more easily copy it into heap space later.
+        let argPtrsArray = Uint32Array.from(
+            args
+                .map((argStr) => {
+                    return Module._malloc(Module.lengthBytesUTF8(argStr) + 1); // +1 for the null terminator byte
+                })
+                .concat([0])
+        );
+        // ^ In addition to each individual argument being null-terminated, the GDAL docs specify that
+        // GDALTranslateOptionsNew takes its options passed in as a null-terminated array of
+        // pointers, so we have to add on a null (0) byte at the end.
+
+        // Next, we need to write each string from the JS string array into the Emscripten heap space
+        // we've allocated for it.
+        args.forEach(function (argStr, i) {
+            Module.stringToUTF8(
+                argStr,
+                argPtrsArray[i],
+                Module.lengthBytesUTF8(argStr) + 1
+            );
+        });
+
+        // Now, as mentioned above, we also need to copy the pointer array itself into heap space.
+        let argPtrsArrayPtr = Module._malloc(
+            argPtrsArray.length * argPtrsArray.BYTES_PER_ELEMENT
+        );
+
+        Module.HEAPU32.set(
+            argPtrsArray,
+            argPtrsArrayPtr / argPtrsArray.BYTES_PER_ELEMENT
+        );
+
+        self.errorHandling = errorHandling;
+        self.argPtrsArray = argPtrsArray;
+        self.argPtrsArrayPtr = argPtrsArrayPtr;
+    }
+
+    validateOptions() {
```

Comment:
Option validation could certainly be handled outside the instance after the parameters are parsed. Almost seems right to extract the common error handling logic in a shareable function between the different parts of the API, but you would know what is best.

Reply:
Yeah, I think having a shareable function for error handling logic is exactly the right way to go, but it's not necessary to do that as part of this PR unless you want to -- I can open another issue for factoring out the error handling. Either way, I don't think that option validation needs to be part of this class; I think it will keep things simpler to have this be a single-purpose class for handling memory allocations, and then create a separate set of functions or a class for the parameter validation.